File: kframework/src/main/java/showmethe/github/kframework/widget/citypicker/bean/CityBean.java
package showmethe.github.kframework.widget.citypicker.bean;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;

import androidx.databinding.ObservableArrayList;


public class CityBean implements Parcelable {
    

    private String id;
    
    private String name;


    private ObservableArrayList<DistrictBean> cityList;

    @Override
    public String toString() {
        return name;
    }

    public String getId() {
        return id == null ? "0" : id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name == null ? "" : name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public ObservableArrayList<DistrictBean> getCityList() {
        return cityList;
    }

    public void setCityList(ObservableArrayList<DistrictBean> cityList) {
        this.cityList = cityList;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.id);
        dest.writeString(this.name);
        dest.writeTypedList(this.cityList);
    }

    public CityBean() {
    }

    protected CityBean(Parcel in) {
        this.id = in.readString();
        this.name = in.readString();
        this.cityList = (ObservableArrayList<DistrictBean>) in.createTypedArrayList(DistrictBean.CREATOR);
    }

    public static final Creator<CityBean> CREATOR = new Creator<CityBean>() {
        @Override
        public CityBean createFromParcel(Parcel source) {
            return new CityBean(source);
        }

        @Override
        public CityBean[] newArray(int size) {
            return new CityBean[size];
        }
    };
}

File: kframework/src/main/java/showmethe/github/kframework/widget/citypicker/bean/CityBean.kt
package showmethe.github.kframework.widget.citypicker.bean

import android.os.Parcel
import android.os.Parcelable
import androidx.databinding.ObservableArrayList

data class CityBean(val id: String = "", var name: String = "",var cityList: ObservableArrayList<DistrictBean>) : Parcelable {
    constructor(source: Parcel) : this(
        source.readString()!!,
        source.readString()!!,
        source.createTypedArrayList(CREATOR) as ObservableArrayList<DistrictBean>
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeString(id)
        writeString(name)
        writeTypedList<DistrictBean>(cityList)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<CityBean> = object : Parcelable.Creator<CityBean> {
            override fun createFromParcel(source: Parcel): CityBean = CityBean(source)
            override fun newArray(size: Int): Array<CityBean?> = arrayOfNulls(size)
        }
    }
}